src/sql/azdata.proposed.d.ts

```diff
@@ -995,24 +995,35 @@ declare module 'azdata' {
 		 */
 		export interface TableDesignerProvider extends DataProvider {
 			/**
-			 * Gets the table designer information for the specified table.
+			 * Initialize the table designer for the specified table.
 			 * @param table the table information.
 			 */
-			getTableDesignerInfo(table: TableInfo): Thenable<TableDesignerInfo>;
+			initializeTableDesigner(table: TableInfo): Thenable<TableDesignerInfo>;
+
 			/**
 			 * Process the table change.
 			 * @param table the table information
-			 * @param viewModel the object contains the state of the table designer
 			 * @param tableChangeInfo the information about the change user made through the UI.
 			 */
-			processTableEdit(table: TableInfo, viewModel: DesignerViewModel, tableChangeInfo: DesignerEdit): Thenable<DesignerEditResult>;
+			processTableEdit(table: TableInfo, tableChangeInfo: DesignerEdit): Thenable<DesignerEditResult>;
+
+			/**
+			 * Publish the changes.
+			 * @param table the table information
+			 */
+			publishChanges(table: TableInfo): Thenable<void>;
+
+			/**
+			 * Generate script for the changes.
+			 * @param table the table information
+			 */
+			generateScript(table: TableInfo): Thenable<string>;
 
 			/**
-			 * Save the table
+			 * Generate preview report for the changes.
```

Comment:
If you plan on leaving this as "report" I'd strongly suggest adding a description of what a "report" is in this context. At least something like `generate preview report describing the changes to be made` or something would help a lot with understanding what to expect.